src/components/landing/pages/login/auth/via-ldap.tsx

```diff
@@ -17,49 +17,45 @@ export const ViaLdap: React.FC = () => {
 
   const name = ldapCustomName ? `${ldapCustomName} (LDAP)` : 'LDAP'
 
-  const doAsyncLogin = async () => {
-    try {
-      await doLdapLogin(username, password)
-      await getAndSetUser()
-    } catch {
-      setError(true)
-    }
-  }
-
-  const onFormSubmit = (event: FormEvent) => {
-    doAsyncLogin().catch(() => setError(true))
+  const onLoginSubmit = useCallback((event: FormEvent) => {
+    doLdapLogin(username, password).then(() => {
+      getAndSetUser().catch(() => setError(true))
+    }).catch(() => setError(true))
```

Comment:
```suggestion
    doLdapLogin(username, password)
      .then(() => getAndSetUser())
      .catch(() => setError(true))
```
Why aren't you using Promise-Chaining?

Reply:
There were some ESLint errors, that irritated me so I built that solution. Turned out it was a caching error and promise chaining works as expected without errors. :D